earthly/rust/Earthfile

```diff
@@ -54,77 +50,232 @@ rust-base:
 
     # Universal build scripts we will always need and are not target dependent.
     COPY --dir scripts /scripts
     # Copy our common scripts so we can use them inside the container.
-    DO ../../utilities/scripts+ADD_BASH_SCRIPTS
-    DO ../../utilities/scripts+ADD_PYTHON_SCRIPTS
+    DO scripts+ADD_BASH_SCRIPTS
+    DO scripts+ADD_PYTHON_SCRIPTS
 
     # Standardized Rust configs.
     # Build will refuse to proceed if the projects rust configs do not match these.
     # This is to enforce consistent compiler and tool configuration on local setup and CI builds.
     COPY --dir stdcfgs /stdcfgs
 
-# Builds all the rust-base targets for each supported DOCKER architecture.
-# Currently only used for multi-platform cross build testing.
-# This will ONLY work if you have `qemu` properly setup on linux and `rosetta` for
-# docker enabled on Mac.
-# Again, this is just a test target, and not for general use.
+# rust-base-all-hosts : Builds all the rust-base targets for each supported DOCKER architecture.
+#   Currently only used for multi-platform cross build testing.
+#   This will ONLY work if you have `qemu` properly setup on linux and `rosetta` for
+#   docker enabled on Mac.
+#   Again, this is just a test target, and not for general use.
 rust-base-all-hosts:
-    BUILD --platform=linux/amd64 --platform=linux/arm64 +rust-base
+    BUILD --platform=linux/amd64 --platform=linux/arm64 +installer
 
-# Common Rust setup.
-# Parameters:
-#  * toolchain : The `rust-toolchain` toml file.
-SETUP:
+# EXECUTE : runs the cargo command "$args".
+# This function is thread safe. Parallel builds of targets calling this function should be free of race conditions.
+# Notice that in order to run this function, +INIT must be called first.
+# Arguments:
+#   - cmd: Command and its arguments. Required.
+#   - output: Regex matching output artifacts files to be copied to ./target folder in the caller filesystem (image layers).
+#     Use this argument when you want to SAVE an ARTIFACT from the target folder (mounted cache), always trying to minimize the total size of the copied fileset.
+#     For example --output="release/[^\./]+" would keep all the files in /target/release that don't have any extension.
+#   - junit: Filename to save the junit.xml file LOCALLY.
+#            Also saves the junit.xml file in the target layer.
+#            If not defined, "junit.xml" will not be saved, and will not be in the layer.
+#   - coverage: Filename to save the coverage-report.info as LOCALLY
+#            If not defined, "coverage-report.info" will not be saved, but will be in the layer.
+#
+# This is heavily based off of the rust `CARGO` command in the Earthly library.
+# Updates to that library must be reflected in this function when updating the version of the earthly rust library.
+EXECUTE:
     FUNCTION
-    FROM +rust-base
+    DO rust+CHECK_INITED
+    ARG --required cmd
+    ARG args1
```

Comment:
Maybe it will be better to compose all arguments and cmd on the clients side and remove all these `arg1`, `arg2` etc.
Because with such approach it could be a case when we need to add more, which does not seems like a best solution.
So on the client's side of the `EXECUTE` function it could be something like this.

```Earthfile
target:
   ...
   LET cmd = "foo"
   LET arg1 = "--arg1=arg1"
   LET arg2 = "--arg2=arg2"
   
   +EXECUTE --cmd="$cmd $arg1 $arg2"
```

Reply:
No, i prefer it the way I did it,  otherwise we push management of this Earthly quirk to all users.  (Ugly is subjective, but I think it is).